csrc/ir_nodes.cpp

```diff
@@ -2500,6 +2500,18 @@ void IterDomain::parallelize(ParallelType t) {
     return;
   }
 
+  // assert check that we only parallelize a leaf domain.
+  // leaf domains are domains that are not used by any other domains.
+  TORCH_CHECK(
```

Comment:
Doesn't this fail when `t == ParallelType::Serial`?

Reply:
In that case, `parallel_type_ == t` is true. After I moved the check after 
```
  if (parallel_type_ == t) {
    // No op, don't do any more checks, it was already set to this value.
    return;
  }
```
then, the check will not be called and we are good.

Reply:
Why is it guaranteed to be true?

Reply:
Note that it is totally fine to remove parallelization of a parallelized IterDomain by `parallelize(ParalleType::Serial)`, so this isn't correct.

> In that case, parallel_type_ == t is true.

Reply:
updated